### PR TITLE
[Web-App] Redesign earn page

### PR DESCRIPTION
**Pull request by Maks19**
**Description:** Redesign earn page

**Notion Project:** https://www.notion.so/saladtech/Frontend-Earning-Page-Design-Refresh-7bc4991e6c5c47168d4450fa257ee9db

image

image

### CODE REVIEW

File: packages/web-app/src/modules/earn-views/components/EarningChart.tsx [hunk 1]
```diff
@@ -61,9 +76,12 @@ const styles = (theme: SaladTheme) => ({
     marginTop: -20,
   },
   tooltipContainer: {
-    fontFamily: theme.fontGroteskBook25,
+    fontFamily: 'Mallory',
```
Comment (vitto-moz) [inline packages/web-app/src/modules/earn-views/components/EarningChart.tsx:79]: wonder why we have to set the `fontFamily` for a particular class, when we have it set globally as `Mallory`

File: packages/web-app/src/modules/earn-views/components/EarningChart.tsx [hunk 2]
```diff
@@ -201,12 +193,16 @@ const _CustomizedXAxisTick = (props: CustomTick) => {
...
+  const isMidnight = moment(payload.value).add(15, 'minute').hours() === 0
+  const isNoon = moment(payload.value).add(15, 'minute').hours() === 12
+  const shouldShowAmPm = daysShowing === 1
+  const shouldShowDateMonth = daysShowing === 7
+
+  const timestamp = shouldShowAmPm
+    ? moment(payload.value).add(15, 'minute').format('h')
+    : shouldShowDateMonth
+    ? moment(payload.value).add(15, 'minute').format('D/M')
+    : moment(payload.value).add(15, 'minute').format('D')
```
Comment (vitto-moz) [inline packages/web-app/src/modules/earn-views/components/EarningChart.tsx:205]: nested ternary is a bad practice - as decreases readability

File: packages/web-app/src/modules/earn-views/components/LatestRewardsRedeemed/components/LatestRewardsRedeemedCard.tsx [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    marginTop: 'auto',
+  },
+})
+
+interface Props extends WithStyles<typeof styles> {
+  id: string
+  name: string
+  coverImage?: string
+  timestamp: Date
+}
```
Comment (vitto-moz) [inline packages/web-app/src/modules/earn-views/components/LatestRewardsRedeemed/components/LatestRewardsRedeemedCard.tsx:59]: why don't we just use `RedeemedReward` type here?

File: packages/web-app/src/modules/vault/VaultStore.tsx [hunk 4]
```diff
@@ -1,16 +1,26 @@
...
+import { RewardVaultStatus } from './models'
 
 export class VaultStore {
   private refreshTimer?: NodeJS.Timeout
 
   @observable
   public redemptions: RewardVaultItem[] = []
 
+  @observable
+  public latestCompletedRedeemedRewards: Map<string, RedeemedReward> = new Map<string, RedeemedReward>()
```
Comment (vitto-moz) [inline packages/web-app/src/modules/vault/VaultStore.tsx:19]: what is the idea of using `Map` here?

File: packages/web-app/src/modules/vault/VaultStore.tsx [hunk 5]
```diff
@@ -31,13 +41,55 @@ export class VaultStore {
...
+      .reverse()
+
+    this.isLatestCompletedRedeemedRewardsLoading = true
+
+    try {
+      for (let completedRedeemedReward of latestCompletedRedeemedRewards) {
+        yield this.addRewardToCompletedRedeemedList(completedRedeemedReward)
+      }
+      this.isLatestCompletedRedeemedRewardsLoading = false
+    } catch {}
```
Comment (vitto-moz) [inline packages/web-app/src/modules/vault/VaultStore.tsx:64]: any error handling here?

File: packages/web-app/src/modules/vault/VaultStore.tsx [hunk 6]
```diff
@@ -31,13 +41,55 @@ export class VaultStore {
...
+    try {
+      const res: AxiosResponse<RewardResource> = yield this.axios.get(
+        `/api/v1/rewards/${completedRedeemedReward.rewardId}`,
+      )
+      const reward: RedeemedReward = this.completedRedeemedRewardFromResource(
+        completedRedeemedReward,
+        res.data.coverImage,
+      )
+      this.latestCompletedRedeemedRewards.set(reward.id, reward)
+    } catch {}
```
Comment (vitto-moz) [inline packages/web-app/src/modules/vault/VaultStore.tsx:78]: any error handling here?

**Review by vitto-moz — CHANGES_REQUESTED** [review]

File: packages/web-app/src/modules/earn-views/components/EarningChart.tsx (hunk 1 above)
Comment (Maks19) [reply packages/web-app/src/modules/earn-views/components/EarningChart.tsx:79]: we're styling elements from third party library so we need `fontFamily` here

File: packages/web-app/src/modules/vault/VaultStore.tsx (hunk 6 above)
Comment (Maks19) [reply packages/web-app/src/modules/vault/VaultStore.tsx:78]: Added console error as we do not need to interrupt other logic or show any error screen

File: packages/web-app/src/modules/earn-views/components/LatestRewardsRedeemed/components/LatestRewardsRedeemedCard.tsx (hunk 3 above)
Comment (Maks19) [reply packages/web-app/src/modules/earn-views/components/LatestRewardsRedeemed/components/LatestRewardsRedeemedCard.tsx]: Fixed

File: packages/web-app/src/modules/earn-views/components/EarningChart.tsx (hunk 2 above)
Comment (Maks19) [reply packages/web-app/src/modules/earn-views/components/EarningChart.tsx]: Agreed
Changed

File: packages/web-app/src/modules/balance/utils.tsx [hunk 7]
```diff
@@ -12,3 +14,24 @@ export const batchEarningsWindow = (earnings: Map<number, number>, batchSize: nu
...
 }
+
+export const getEarningWindowsGroupedByDay = (
+  earningWindows: EarningWindow[],
+  numberOfDays: 7 | 30,
+): EarningWindow[] => {
+  const groupedByTheDayEarningWindows = groupBy(earningWindows, (item) => moment(item.timestamp).endOf('day').unix())
+
+  const accumulatedEarningsByTheDay = Object.values(groupedByTheDayEarningWindows).map((item) =>
+    item.reduce((acc, curr) => ({
```
[the remark this answers is not in the document]
Comment (Maks19) [reply packages/web-app/src/modules/balance/utils.tsx]: Days
Changed to `day`

File: packages/web-app/package.json
Comment (Maks19) [inline packages/web-app/package.json:1]: Update garden-components version when the build is fixed and this [PR ](https://github.com/SaladTechnologies/salad-applications/pull/1002) will be merged

**Comment by Maks19** [comment]
Open new PR with fixed build https://github.com/SaladTechnologies/salad-applications/pull/1022

**Pull request closed without merging**